File: listProject/ruby/link.rb
class Link 
  def initialize(data, nexts = nil) 
    @data = data
    @next = nexts
  end

  def getChain() 
    puts getData()
    if getNext() 
      getNext().getChain()
    end
  end

  def getData() 
    return @data;
  end

  def getNext() 
    return @next;
  end

  def setData(data) 
    return @data = data
  end

  def setNext(nexts) 
    return @next = nexts
  end

end

File: listProject/javascript/link.js
export default class Link {
  constructor(data, next = null) {
    this.data = data;
    this.next = next;
  }

  getChain() {
    console.log(this.getData());
    if (this.getNext()) {
      this.getNext().getChain();
    }
  }

  getData() {
    return this.data;
  }

  getNext() {
    return this.next;
  }

  // getPos() {
  //   return this.pos;
  // }
  3;

  setData(data) {
    return (this.data = data);
  }

  setNext(next) {
    return (this.next = next);
  }

  // setPos(pos) {
  //   return (this.pos = pos);
  // }
}
